### PR TITLE
Allow APITestCase and ModelViewSet (etc.) to work with plugin models

### PR DESCRIPTION
**Pull request by glennmatthews**
### Fixes: #336 (thanks @lampwins)

A few more spot fixes for plugin usability of core code - these:

1. Make it possible to automatically look up a plugin's `Nested*Serializer`, if available, to allow a plugin REST API endpoint (`ModelViewSet`) to easily support `brief=true` like the core REST API endpoints do.
2. Allow APITestCase to be used to test plugin REST API endpoints without needing to explicitly specify `view_namespace` in each test case.

### CODE REVIEW

**Comment by lampwins** [comment]
It seems this addresses #336

File: nautobot/utilities/api.py [hunk 1]
```diff
@@ -19,7 +19,9 @@ def get_serializer_for_model(model, prefix=""):
     # Serializers for Django's auth models are in the users app
     if app_name == "auth":
         app_name = "users"
-    serializer_name = f"nautobot.{app_name}.api.serializers.{prefix}{model_name}Serializer"
+    serializer_name = f"{app_name}.api.serializers.{prefix}{model_name}Serializer"
```
Comment (FragmentedPacket) [inline nautobot/utilities/api.py:22]: I was thinking that we could potentially move this into a function within `nautobot/utilities` as it seems like we may be reusing it more in the future.

Doesn't have to be in this PR, but maybe a separate issue to track?

Comment (glennmatthews) [reply nautobot/utilities/api.py:22]: So far (this PR, #357, #303, #293) in each case we've checked for `app_name in settings.PLUGINS` but the logic that we've needed to conditionally execute in each case has been different. Consolidation of reused logic is absolutely something I'm in favor of but so far I'm not seeing what that might look like in this particular case.

Comment (FragmentedPacket) [reply nautobot/utilities/api.py:22]: Good point!

**Review by FragmentedPacket — APPROVED** [review]

**Comment by glennmatthews** [comment]
Also fixes #399 now.

**Review by jathanism — APPROVED** [review]
🙈

**Pull request merged**